app/src/ui/branches/PushBranchCommits.tsx

```diff
@@ -54,98 +89,98 @@ export class PushBranchCommits extends React.Component<
         onDismissed={this.cancel}
         onSubmit={this.cancel}
+        loading={this.state.loading}
       >
-        <DialogContent>
-          {this.renderDialogContent()}
-        </DialogContent>
+        {this.renderDialogContent()}
 
         <DialogFooter>
-          <ButtonGroup destructive={true}>
-            <Button type="submit">Cancel</Button>
-            <Button onClick={this.push}>
-              {this.renderButtonText()}
-            </Button>
-          </ButtonGroup>
+          {this.renderButtonGroup()}
         </DialogFooter>
       </Dialog>
     )
   }
 
   private renderDialogContent() {
-    if (this.isPublish) {
+    if (renderPublishView(this.props.unPushedCommits)) {
       return (
-        <p>
-          Your branch must be published before opening a pull request. Would you
-          like to publish <b>{this.props.branch.name}</b> and open a pull
-          request?
-        </p>
+        <DialogContent>
+          <p>Your branch must be published before opening a pull request.</p>
+          <p>
+            Would you like to publish <Ref>{this.props.branch.name}</Ref> now
+            and open a pull request?
+          </p>
+        </DialogContent>
       )
     }
 
-    const numberOfCommits = this.props.unPushedCommits
+    const localCommits = pluralize(this.props.unPushedCommits, 'local commit')
 
     return (
-      <p>
-        {`Would you like to push ${numberOfCommits} ${this.createCommitString()} to `}
-        <b>{this.props.branch.name}</b> and open a pull request?
-      </p>
+      <DialogContent>
+        <p>
+          You have {localCommits} that haven't been pushed to the remote yet.
+        </p>
+        <p>
+          Would you like to push your changes to{' '}
+          <Ref>{this.props.branch.name}</Ref> before creating your pull request?
+        </p>
+      </DialogContent>
     )
   }
 
   private renderDialogTitle() {
-    if (this.isPublish) {
-      return __DARWIN__ ? 'Publish Branch' : 'Publish branch'
+    if (renderPublishView(this.props.unPushedCommits)) {
+      return __DARWIN__ ? 'Publish Branch?' : 'Publish branch?'
     }
 
-    const numberOfCommits = this.props.unPushedCommits
-
-    return __DARWIN__
-      ? `Your Branch is Ahead by ${numberOfCommits} ${this.createCommitString(
-          true
-        )}`
-      : `Your branch is ahead by ${numberOfCommits} ${this.createCommitString(
-          true
-        )}`
+    return __DARWIN__ ? `Push Local Changes?` : `Push local changes?`
   }
 
-  private renderButtonText() {
-    if (this.isPublish) {
-      return __DARWIN__
-        ? 'Publish Branch and Open Pull Request'
-        : 'Publish branch and open pull request'
+  private renderButtonGroup() {
+    if (renderPublishView(this.props.unPushedCommits)) {
+      return (
+        <ButtonGroup destructive={true}>
+          <Button type="submit">Cancel</Button>
+          <Button onClick={this.onPushOrPublishButtonClick}>Yes</Button>
```

Comment:
On Mac they recommend against "Yes" as a button title and recommend using the action as the affirmative answer.